utils_test.go

```diff
@@ -272,6 +280,7 @@ func TestRecordMetric(t *testing.T) {
 		args           args
 		recordErr      error
 		expectedStatus string
+		expectedAttr   attribute.KeyValue
```

Comment:
I prefer we compare the entire attribute set directly.
```suggestion
		expectedAttrs   attribute.Set
```

Then, construct and compare attributes directly

```go
expectedAttrs: attribute.NewSet(attribute.String("method", "sql.conn.query"), attribute.String("status", "ok")
```

```go
assert.Equal(t, tt.expectedAttrs, mockLatency.attrs)
```